modules/auxiliary/scanner/x11/open_x11.rb

```diff
@@ -30,47 +41,23 @@ def initialize
   end
 
   def run_host(ip)
+    connect
+    connection = x11_connect
 
-    begin
-
-      connect
-
-      # X11.00 Null Auth Connect
-      sock.put("\x6c\x00\x0b\x00\x00\x00\x00\x00\x00\x00\x00\x00")
-      response = sock.get_once
-
-      disconnect
-
-      if response
-        success = response[0,1].unpack('C')[0]
-      else
-        print_error("No response received due to a timeout")
-        return
-      end
-
-
-      if(success == 1)
-        vendor_len = response[24,2].unpack('v')[0]
-        vendor = response[40,vendor_len].unpack('A*')[0]
-        print_good("#{ip} Open X Server (#{vendor})")
-        # Add Report
-        report_note(
-          :host	=> ip,
-          :proto => 'tcp',
-          :sname	=> 'x11',
-          :port	=> rport,
-          :type	=> 'Open X Server',
-          :data	=> "Open X Server (#{vendor})"
-      )
-      elsif (success == 0)
-        print_error("#{ip} Access Denied")
-      else
-        # X can return a reason for auth failure but we don't really care for this
-      end
+    if connection.nil?
+      vprint_bad('No connection, or bad X11 response received')
+      return
+    end
 
-    rescue ::Rex::ConnectionError
-    rescue ::Errno::EPIPE
+    if connection.header.success == 1
+      print_connection_info(connection, ip, rport)
```

Comment:
This doesn't report it as a service but as a note. The scanner here should definitely report it as a service since that's probably the context in which this module would be used. I'd either recommend updating `#print_connection_info` to report the service, or report it here.